### ISSUE TITLE
Side-loading on GearVR not Working

### ISSUE THREAD

**Issue by felcerreia** [GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR#26]
Hi, I followed all the steps for sideloading the app in the Gear VR on a Galaxy S7, but when I dock the phone it opens the default home app.

I tested with the test app that comes with the Injector and everything worked. So I think there is something wrong with the APK.

**Comment by GennadiyAMD** [comment]
@felcerreia  - just tried it myself and everything worked as expected on my Galaxy S7. When the APK doesn't have the Oculus signature for your device (this is what the Injector app adds), the phone displays the following message once you dock it in GearVR: "Thread priority security exception. Make sure the APK is signed." and only then, once you click OK, it takes you to the default Oculus Home app. Are you getting the same message or is it taking you straight to the Oculus Home app?

**Comment by felcerreia** [comment]
@GennadiyAMD I'm using the injector, when I execute the ReLive app this shows a screen asking to connect the gear vr, after that the phone goes directly to the Home app =/

Tested with the demo/test app, and the quakevr and they just work fine.

I don't have the control pad, do you think this could be causing the problem?

BTW I'm using this [apk ](https://github.com/GPUOpen-LibrariesAndSDKs/Radeon-ReLive-VR/raw/master/ReLiveVR-Oculus-1.0.18.apk) with the injector

**Comment by felcerreia** [comment]
I just managed to get some log.
`11-10 01:02:40.177  9461  9461 E Zygote  : isWhitelistProcess - Process is Whitelisted
11-10 01:02:40.179  9461  9461 E libpersona: scanKnoxPersonas
11-10 01:02:40.179  9461  9461 E libpersona: Couldn't open the File - /data/system/users/0/personalist.xml - No such file or directory
11-10 01:02:40.185  9461  9461 W SELinux : SELinux selinux_android_compute_policy_index : Policy Index[2],  Con:u:r:zygote:s0 RAM:SEPF_SM-G930F_8.0.0_0011, [-1 -1 -1 -1 0 1]
11-10 01:02:40.186  9461  9461 I SELinux : SELinux: seapp_context_lookup: seinfo=untrusted, level=s0:c512,c768, pkgname=com.amd.wirelesshmd 
11-10 01:02:40.231  9461  9461 D TimaKeyStoreProvider: TimaKeyStore is not enabled: cannot add TimaSignature Service and generateKeyPair Service
11-10 01:02:40.232  9461  9461 D ActivityThread: Added TimaKeyStore provider
11-10 01:02:40.347  9461  9461 I zygote64: no shared libraies, dex_files: 1
11-10 01:02:40.498  9461  9461 D AMF_Main_WVR: OnCreate()
11-10 01:02:40.532  9461  9461 D AMF_Main_WVR: OnStart()
11-10 01:02:40.533  9461  9461 D AMF_Main_WVR: Intent received: Action=android.intent.action.MAIN, type: null
11-10 01:02:40.595  9461  9461 D AMF_Main_WVR: Unique Device ID: cd6e46ff4adb3e5c
11-10 01:02:40.596  9461  9461 D AMF_TRACE: 0119-10-10 01:02:40 1BABA9C8 [Settings]    Info: Attempting to load client configuration from /storage/emulated/0/Android/data/com.amd.wirelesshmd/files/app.settings
11-10 01:02:40.596  9461  9461 D AMF_TRACE: 0119-10-10 01:02:40 1BABA9C8 [Settings]   Debug: Current Config: Discovery: TRUE, Reconnect: TRUE, Datagram size: 65507
11-10 01:02:40.596  9461  9461 D AMF_TRACE: 0119-10-10 01:02:40 1BABA9C8 [jni.cpp]    Info: Loading native code...
11-10 01:02:40.596  9461  9461 D AMF_TRACE: 0119-10-10 01:02:40 1BABA9C8 [Motor]    Info: Init
11-10 01:02:40.604  9461  9461 D AMF_Main_WVR: OnResume()
11-10 01:02:40.619  9461  9482 D OpenGLRenderer: HWUI GL Pipeline
11-10 01:02:40.631  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: setView = DecorView@3fba078[MainActivity] TM=true MM=false
11-10 01:02:40.632  9461  9461 I AMF_Main_WVR: WiFi state changed to CONNECTED
11-10 01:02:40.632  9461  9461 I AMF_Main_WVR: Battery level: 58.0%
11-10 01:02:40.634  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: dispatchAttachedToWindow
11-10 01:02:40.644  9461  9461 V Surface : sf_framedrop debug : 0x4f4c, game : false, logging : 0
11-10 01:02:40.645  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: Relayout returned: old=[0,0][0,0] new=[0,0][2560,1440] result=0x7 surface={valid=true 493808357376} changed=true
11-10 01:02:40.649  9461  9482 I OpenGLRenderer: Initialized EGL, version 1.4
11-10 01:02:40.649  9461  9482 D OpenGLRenderer: Swap behavior 2
11-10 01:02:40.653  9461  9482 D libGLESv1: STS_GLApi : DTS, ODTC are not allowed for Package : com.amd.wirelesshmd
11-10 01:02:40.656  9461  9482 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, egl_color_buffer_format *, EGLBoolean) returns 0x3000,  [2560x1440]-format:1
11-10 01:02:40.656  9461  9482 D OpenGLRenderer: eglCreateWindowSurface = 0x730bc28640
11-10 01:02:40.672  9461  9461 D SurfaceView: BG show() Surface(name=Background for - SurfaceView - com.amd.wirelesshmd/com.amd.wirelesshmd.MainActivity@c5cbcb7@0) android.view.SurfaceView{c5cbcb7 V.E...... ......ID 0,0-2560,1440}
11-10 01:02:40.673  9461  9461 V Surface : sf_framedrop debug : 0x4f4c, game : false, logging : 0
11-10 01:02:40.673  9461  9461 V Surface : sf_framedrop debug : 0x4f4c, game : false, logging : 0
11-10 01:02:40.674  9461  9461 D SurfaceView: surfaceCreated 1 android.view.SurfaceView{c5cbcb7 V.E...... ......ID 0,0-2560,1440}
11-10 01:02:40.674  9461  9461 D AMF_Main_WVR: surfaceCreated()
11-10 01:02:40.674  9461  9461 D SurfaceView: surfaceChanged (2560,1440) 1 android.view.SurfaceView{c5cbcb7 V.E...... ......ID 0,0-2560,1440}
11-10 01:02:40.674  9461  9461 D AMF_Main_WVR: surfaceChanged()
11-10 01:02:40.674  9461  9495 D AMF_Main_WVR: RenderThread started
11-10 01:02:40.675  9461  9495 D AMF_TRACE: 0119-10-10 01:02:40 F89204F0 [GearVRRenderer]    Info: Init()
11-10 01:02:40.675  9461  9495 D AMF_TRACE: 0119-10-10 01:02:40 F89204F0 [BaseRenderer]    Info: Init()
11-10 01:02:40.675  9461  9495 I VrApiLoader: VrApi Loader Version: 1.1.24.0-0-0 Jul  8 2019 12:23:35 
11-10 01:02:40.676  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: MSG_RESIZED_REPORT: frame=Rect(0, 0 - 2560, 1440) ci=Rect(0, 0 - 0, 0) vi=Rect(0, 96 - 0, 0) or=2
11-10 01:02:40.676  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: MSG_WINDOW_FOCUS_CHANGED 1
11-10 01:02:40.686  9461  9495 I zygote64: no shared libraies, dex_files: 1
11-10 01:02:40.686  9461  9495 I zygote64: Failed to open app image /data/app/com.oculus.systemdriver-fCcALl1KXIWryJKoIIvTtQ==/oat/arm64/base.art Failed to open '/data/app/com.oculus.systemdriver-fCcALl1KXIWryJKoIIvTtQ==/oat/arm64/base.art'
11-10 01:02:40.707  9461  9461 V InputMethodManager: Starting input: tba=android.view.inputmethod.EditorInfo@486f942 nm : com.amd.wirelesshmd ic=null
11-10 01:02:40.707  9461  9461 I InputMethodManager: startInputInner - mService.startInputOrWindowGainedFocus
11-10 01:02:40.710  9461  9495 V VrDriver: Loading 64-bit VrDriver.
11-10 01:02:40.715  9461  9461 D ViewRootImpl@1c41ddb[MainActivity]: Relayout returned: old=[0,0][2560,1440] new=[0,0][2560,1440] result=0x3 surface={valid=true 493808357376} changed=false
**11-10 01:02:40.723  9461  9495 W System.err: java.lang.RuntimeException: Requested API version (api=1.1.24.0, driver=0) is incompatible with the currently installed version.
11-10 01:02:40.723  9461  9495 W System.err: 	at com.oculus.systemdriver.DriverLoader.loadInternal(DriverLoader.java:120)
11-10 01:02:40.723  9461  9495 W System.err: 	at com.oculus.systemdriver.DriverLoader.load64(DriverLoader.java:279)
11-10 01:02:40.723  9461  9495 W System.err: 	at com.amd.wirelesshmd.WirelessHMDClient.initRenderer(Native Method)
11-10 01:02:40.723  9461  9495 W System.err: 	at com.amd.wirelesshmd.MainActivity$2.run(MainActivity.java:650)
11-10 01:02:40.723  9461  9495 W System.err: 	at java.lang.Thread.run(Thread.java:764)
11-10 01:02:40.723  9461  9495 W VrApiLoader: vrapi_Initialize: Failed to load VrApi driver! Java Exception**
11-10 01:02:40.776  9461  9461 D AMF_Main_WVR: OnPause()
11-10 01:02:40.874  9461  9461 D SurfaceView: BG show() Surface(name=Background for - SurfaceView - com.amd.wirelesshmd/com.amd.wirelesshmd.MainActivity@c5cbcb7@0) android.view.SurfaceView{c5cbcb7 V.E...... ........ 0,0-2560,1440}
11-10 01:02:40.877  9461  9461 D SurfaceView: surfaceDestroyed 1 android.view.SurfaceView{c5cbcb7 V.E...... ........ 0,0-2560,1440}
11-10 01:02:40.877  9461  9461 D AMF_Main_WVR: surfaceDestroyed()
11-10 01:02:40.878  9461  9461 D AMF_Main_WVR: RenderThreadShutdown started
11-10 01:02:40.890  9461  9495 I zygote64: System.exit called, status: 0
11-10 01:02:40.890  9461  9495 I AndroidRuntime: VM exiting with result code 0, cleanup skipped.`

**Comment by GennadiyAMD** [comment]
@felcerreia - looks like you have an outdated GearVR software from Oculus. Check which version you have:
Go to Android Settings->Apps
Scroll down to find Gear VR Service, tap on it
Scroll down to see which version you have. Mine is reporting 3.2.02.13 and ReLive VR seems to be happy with that.

ReLive VR uses the latest Oculus SDK 1.24.0, which seems to require the version of GearVR runtime which is newer than what you seem to have.

**Comment by felcerreia** [comment]
@GennadiyAMD I checked and was in the same version.

I think I managed to solve the problem uninstalling all apps with Oculus and GearVR in the name and them reopening the GearVR app just letting it reinstall everything. Now it's showing an image from the controller pad with a red cross, I think it's normal because I don't have the controller paired yet.

Thanks for the help figuring out how to solve the problem!
